tests/parametric/test_headers_tracecontext.py

```diff
@@ -807,6 +807,165 @@ def test_tracestate_w3c_p_inject(self, test_agent, test_library):
+    @missing_feature(context.library == "python", reason="Not implemented")
+    @missing_feature(context.library == "dotnet", reason="Not implemented")
+    @missing_feature(context.library == "php", reason="Not implemented")
+    @missing_feature(context.library == "nodejs", reason="Not implemented")
+    @missing_feature(context.library == "java", reason="Not implemented")
+    @missing_feature(context.library == "cpp", reason="Not implemented")
+    @missing_feature(context.library == "ruby", reason="Not implemented")
+    @missing_feature(context.library == "golang", reason="Not implemented")
+    @pytest.mark.parametrize("library_env", [{"DD_TRACE_PROPAGATION_STYLE": "datadog,tracecontext"}])
+    def test_tracestate_w3c_p_phase_3_extract_inject(self, test_agent, test_library):
+        """
+        Ensure the last parent id tag is set according to the W3C phase 3 spec
+        """
+        with test_library:
+
+            # 1) Datadog and tracecontext headers, trace ids and parent ids match
+            with test_library.start_span(
+                name="matching_trace_id_and_span_id",
+                http_headers=[
+                    ["traceparent", "00-11111111111111110000000000000001-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-trace-id", "1"],
+                    ["x-datadog-parent-id", "987654321"],
+                ],
+            ):
+                pass
+
+            # 2) Datadog and tracecontext headers, trace ids and parent ids match, missing p
+            with test_library.start_span(
+                name="matching_trace_id_and_span_id_missing_p",
+                http_headers=[
+                    ["traceparent", "00-11111111111111110000000000000002-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2,foo=1"],
+                    ["x-datadog-trace-id", "2"],
+                    ["x-datadog-parent-id", "987654321"],
+                ],
+            ):
+                pass
+
+            # 3) Datadog and tracecontext headers, only trace ids matches
+            with test_library.start_span(
+                name="matching_trace_id_only",
+                http_headers=[
+                    ["traceparent", "00-11111111111111110000000000000003-000000003ade68b1-01"],
+                    ["tracestate", "dd=s:2;p:0123456789abcdef,foo=1"],
+                    ["x-datadog-trace-id", "3"],
+                    ["x-datadog-parent-id", "9876543210"],
+                ],
+            ):
+                pass
+
+            # 4) Datadog headers only
+            with test_library.start_span(
+                name="only_datadog_headers",
+                http_headers=[["x-datadog-trace-id", "4"], ["x-datadog-parent-id", "987654321"],],
+            ):
+                pass
+
+            # 5) Datadog and tracecontext headers, no x-datadog-trace-id, only tid DD tag matches
```

Comment:
`no x-datadog-trace-id`

Looks like there is a `x-datadog-trace-id`, but that it doesn't match the trace ID in `traceparent`

`only tid DD tag matches`

I don't think it does - the `tid` is the upper 64 bits, but the upper 64 bits of the traceparent aren't `5`

I'd recommend changing these to be more explicit - maybe making the `x-datadog-trace-id` not match and making the `tid` match, which should still have the same results as what this one is currently testing.